test/e2e-go/features/transactions/proof_test.go

```diff
@@ -93,37 +95,111 @@ func TestTxnMerkleProof(t *testing.T) {
+	proofrespSHA256, err := client.TxnProof(txid.String(), confirmedTx.ConfirmedRound, crypto.Sha256)
 	a.NoError(err)
 
-	var proof merklearray.Proof
-	proof.HashFactory = crypto.HashFactory{HashType: hashtype}
-	proof.TreeDepth = uint8(proofresp.Treedepth)
-	a.NotEqual(proof.TreeDepth, 0)
-	proofconcat := []byte(proofresp.Proof)
-	for len(proofconcat) > 0 {
-		var d crypto.Digest
-		copy(d[:], proofconcat)
-		proof.Path = append(proof.Path, d[:])
-		proofconcat = proofconcat[len(d):]
+	generateProof := func(h crypto.HashType, prfRsp generated.ProofResponse) (p merklearray.Proof) {
+		p.HashFactory = crypto.HashFactory{HashType: h}
+		p.TreeDepth = uint8(prfRsp.Treedepth)
+		a.NotEqual(p.TreeDepth, 0)
+		proofconcat := prfRsp.Proof
+		for len(proofconcat) > 0 {
+			var d crypto.Digest
+			copy(d[:], proofconcat)
+			p.Path = append(p.Path, d[:])
+			proofconcat = proofconcat[len(d):]
+		}
+		return
 	}
 
-	blk, err := client.BookkeepingBlock(confirmedTx.ConfirmedRound)
-	a.NoError(err)
+	proof := generateProof(crypto.Sha512_256, proofresp)
+	proofSHA256 := generateProof(crypto.Sha256, proofrespSHA256)
 
 	element := TxnMerkleElemRaw{Txn: crypto.Digest(txid)}
 	copy(element.Stib[:], proofresp.Stibhash[:])
 
 	elems := make(map[uint64]crypto.Hashable)
 
 	elems[proofresp.Idx] = &element
-	err = merklearray.Verify(blk.TxnRoot.ToSlice(), elems, &proof)
+	err = merklearray.Verify(blk.TxnRoot.DigestSha512_256.ToSlice(), elems, &proof)
 	if err != nil {
-		t.Logf("blk.TxnRoot : %v \nproof path %v \ndepth: %d \nStibhash %v\nIndex: %d", blk.TxnRoot.ToSlice(), proof.Path, proof.TreeDepth, proofresp.Stibhash, proofresp.Idx)
+		t.Logf("blk.TxnRoot : %v \nproof path %v \ndepth: %d \nStibhash %v\nIndex: %d", blk.TxnRoot.DigestSha512_256.ToSlice(), proof.Path, proof.TreeDepth, proofresp.Stibhash, proofresp.Idx)
+		a.NoError(err)
+	}
+
+	element = TxnMerkleElemRaw{Txn: crypto.Digest(txid)}
+	copy(element.Stib[:], proofrespSHA256.Stibhash[:])
+
+	elems = make(map[uint64]crypto.Hashable)
+
+	elems[proofrespSHA256.Idx] = &element
+	err = merklearray.VerifyVectorCommitment(blk.TxnRoot.DigestSha256.ToSlice(), elems, &proofSHA256)
+	if err != nil {
+		t.Logf("blk.TxnRoot : %v \nproof path %v \ndepth: %d \nStibhash %v\nIndex: %d", blk.TxnRoot.DigestSha256.ToSlice(), proofSHA256.Path, proofSHA256.TreeDepth, proofrespSHA256.Stibhash, proofrespSHA256.Idx)
+		a.NoError(err)
+	}
+}
+
+// make sure transaction proof SHA256 does not exist in current consensus version (until we upgrade)
+func TestTxnMerkleProofSHA256(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	defer fixtures.ShutdownSynchronizedTest(t)
+
+	t.Parallel()
+	a := require.New(fixtures.SynchronizedTest(t))
+
+	var fixture fixtures.RestClientFixture
+	// TODO Stateproof: when adding SHA256 header to the current consensus version, use the template of last version not supporting it instead of this one.
+	fixture.Setup(t, filepath.Join("nettemplates", "TwoNodes50Each.json"))
```

Comment:
use `TwoNodes50EachV30.json` ?

Reply:
Can we assume that this feature will be added to the next consensus upgrade? I want this test to use the latest consensus version not supporting SHA256 header (currently V31) to make sure that this upgrade wouldn't break anything.

Reply:
Ah, I see. Then we have `TwoNodes50EachFuture.json` to guarantee having all new protocol flags on even after the release.